lib/block-supports/layout.php

```diff
@@ -74,12 +73,12 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 				// Handle negative margins for alignfull children of blocks with custom padding set.
 				// They're added separately because padding might only be set on one side.
 				if ( isset( $block_spacing_values['declarations']['padding-right'] ) ) {
-					$padding_right = $block_spacing_values['declarations']['padding-right'];
-					$style        .= "$selector > .alignfull { margin-right:calc($padding_right * -1); }";
+					$padding_right                             = $block_spacing_values['declarations']['padding-right'];
+					$layout_styles[ "$selector > .alignfull" ] = array( 'margin-right' => "calc($padding_right * -1)" );
 				}
 				if ( isset( $block_spacing_values['declarations']['padding-left'] ) ) {
-					$padding_left = $block_spacing_values['declarations']['padding-left'];
-					$style       .= "$selector > .alignfull { margin-left: calc($padding_left * -1); }";
+					$padding_left                              = $block_spacing_values['declarations']['padding-left'];
+					$layout_styles[ "$selector > .alignfull" ] = array( 'margin-left' => "calc($padding_left * -1)" );
```

Comment:
This line overrides the `margin-right` rule so that only `margin-left` is applied.

| Trunk | This PR |
| --- | --- |
| <img width="429" alt="image" src="https://user-images.githubusercontent.com/14988353/181429495-eed0be18-8056-4021-a8a4-b89c8d7050f7.png"> | <img width="428" alt="image" src="https://user-images.githubusercontent.com/14988353/181429573-996312da-0bca-4f7b-a4aa-668bda8c455d.png"> |

Reply:
🤦 Thanks for catching this stuff. 

I'll look into a different way to assemble the styles here.

Reply:
@ramonjd I just pushed a commit fixing these issues 👍

Reply:
reverted fix... I rushed into it and did it wrong